src/utilstrencodings.cpp

```diff
@@ -123,6 +125,75 @@ vector<unsigned char> ParseHex(const string& str)
     return ParseHex(str.c_str());
 }
 
+string EncodeAscii85(const char* istr, size_t len)
+{
+    
+    string retVal; //Default is empty-string
+    if (NULL != istr)
+    {
+        size_t isz = strlen(istr);
+        int32_t olen = ascii85_get_max_encoded_length(isz);
+        
+        if (olen < 0)
+        {
+            return retVal; //Encode size error
+        }
+        else
+        {
+            uint8_t obuf[olen];
+            olen = encode_ascii85((uint8_t *)istr, isz, obuf, olen);
+            if(0 < olen)
+            {
+                std::ostringstream ss;
+                std::copy(obuf, obuf+olen, std::ostream_iterator<uint8_t>(ss, ""));
+                retVal = ss.str();
+            }
+        }
+    }
+    return retVal;
+}
+
+string EncodeAscii85(const string& str)
+{
+    return EncodeAscii85((const char*)str.c_str(), str.size());
+}
+
+vector<unsigned char> DecodeAscii85(const char* ostr, bool* pfInvalid)
+{
+    vector<unsigned char> retVal;
+
+    if (NULL != ostr)
+    {
+        size_t osz = strlen(ostr);
+        int32_t olen = ascii85_get_max_decoded_length(osz);
+        if (olen < 0)
+        {
+            *pfInvalid = true;//Decode size error
+            return retVal;
+        }
+        else
+        {
+            uint8_t dbuf[olen];
+            int32_t ilen = decode_ascii85((uint8_t *)ostr, osz, dbuf, olen);
+
+            if (ilen < 0)
+            {
+                *pfInvalid = true;//Decode error
+                return retVal;
+            }
+            retVal = std::vector<unsigned char>(dbuf, dbuf + ilen);     
+        }
+    }
+
+    return retVal;
+}
+
+string DecodeAscii85(const string& str)
+{
+    vector<unsigned char> vchRet = DecodeAscii85(str.c_str());
+    return (vchRet.size() == 0) ? string() : string((const char*)&vchRet[0], vchRet.size());
+}
+
 string EncodeBase64(const unsigned char* pch, size_t len)
 {
     static const char *pbase64 = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
```

Comment:
use constexpr for all const variables - evaluated at compile-time, not run-time.
static auto pbase64 = "....

Reply:
Rest comment is fine and thank you for the valuable information, i learn day by day :) . But this one i might not get the point. You mean put the 0 from the  `'size() == 0' `comparison to `static const int zero_size = 0` or you mean the whole return value shall be in `constexpr()` ?

Reply:
No, look. If you declare any constant now in modern c++, you don't have to do it the old way const char * szStr = "my string"; This was evaluated by compiler at run-time. Now for constant strings you can use constexpr auto szStr = "my string"; In this case it is evaluated at compile-time. Compiler automatically here defines type as const char *. Also, for any other static const variables and defines you can also use constexpr. Instead of #define MYVAR 5, you can say constexpr size_t MYVAR = 5; Benefits are again - compile time and definite var type. constexpr can be used for much more like functions, structures etc...

Reply:
Most probably I'm stupid AF, but i still do not get the point - in this code snippet you highlighted here, what shall i constexpr ? :(  
(Every decode function in that file looks like exactly the same.)